query-node/joystream-query-node/schema.graphql

```diff
@@ -1,21 +1,291 @@
+type Worker @entity {
+	member: Member!
+
+	"Account used to authenticate in this role"
+	roleAccountId: Bytes!
+
+	# "Whether the role has recurring reward, and if so an identifier for this"
+	# rewardRelationship: Option<RewardRelationshipId>
+
+	# "When set, describes role stake of the worker/lead"
+	# roleStakeProfile: Option<RoleStakeProfile<StakeId, BlockNumber>>
+}
+
+type Curator @entity {
+	"Account used to authenticate"
+	worker: Worker!
+
+	groups: [CuratorGroup] @derivedFrom(field: "curators")
+}
+
+"A group, that consists of 'curators' set"
+type CuratorGroup @entity {
+	"Runtime identifier for group"
+	groupId: BigInt!
+
+	"Activity stats of group"
+	isActive: Boolean!
+
+	"All member curators of this group"
+	curators: [Curator!]
+
+	"All classes this group maintains"
+	maintainerOfClasses: [Class!] @derivedFrom(field: "maintainers")
+}
+
+"A voucher for 'Entity' creation"
+type EntityCreationVoucher @entity {
+	"Class to which voucher applies"
+	class: Class!
+
+	"Controller to which this voucher applies"
+	controller: EntityController!
+
+	"How many are allowed in total"
+	maximumEntitiesCount: BigInt!
+
+	"How many have currently been created"
+	entitiesCreated: BigInt!
+}
+
+enum EntityControllerType {
+	LEAD
+	MEMBER
+	MAINTAINERS
+}
+
+"Owner of an 'Entity'"
+type EntityController @entity {
+	"Current controller, which is initially set based on who created entity"
+	controllerType: EntityControllerType!
+
+	"Optional field, will be set when EntityControllerType is MEMBER"
+	controllerWhenMember: Member
+}
+
+"""
+Represents a particular instance of a class,
+hence each entity corresponds to exactly one class
+"""
+type Entity @entity {
+	"Runtime identifier for entity"
+	entityID: BigInt!
+
+	"Current controller, which is initially set based on who created entity"
+	controller: EntityController!
+
+	"Forbid groups to mutate any property value"
+	isFrozen: Boolean!
+
+	"Prevent from being referenced by any entity (including self-references)"
+	isReferenceable: Boolean!
+
+	"The class of this entity"
+	class: Class!
+
+	"""
+	What schemas under which this entity of a class is available, think
+	v.2.0 Person schema for John, v3.0 Person schema for John
+	"""
+	supportedSchemas: [Schema!]
+
+	"Values for properties on class that are used by some schema used by this entity!"
+	values: [PropertyValue!] @derivedFrom(field: "property")
+
+	"Total number of inbound references from another entities"
+	totalNumberOfInboundReferences: BigInt!
+
+	"Number of inbound references from another entities with 'SameOwner' flag set"
+	sameOwnerInboundReferences: BigInt!
+}
+
+"Fields encode a flattened representation of non-vector properties, only one will be non-null"
+type SingleValue @variant {
+	bool: Boolean
+
+	integer: BigInt
+
+	text: Bytes
+
+	hash: Bytes
+
+	reference: Entity
+}
+
+"""
+Fields encode a flattened representation of vector properties, only one will be non-null
+"""
+type VectorValue @variant {
```

Comment:
Don't sure, if i fully understand how `@variant` directive works. 
Though i concerned about the `nonce: BigInt!` field here. 
We should have this field for each of `VectorValue` variants.

Reply:
Yes, we will have `nonce` (non-null) field for each variants. You can find more on `@variant` directive there https://dzhelezov.gitbook.io/hydra-docs/docs/schema-spec/variant-types

Reply:
ok, great